### ISSUE TITLE
DocumentsProvider sometimes failes to create file

### ISSUE THREAD

**Issue by flocke** [nextcloud/android#8201]
Sometimes when trying to use Nextcloud as a DocumentsProvider in my app ([andOTP](https://github.com/andOTP/andOTP)) I get an error during the creation of the file (see logcat below).
I haven't found a way to reliably reproduce this issue yet, it just seems to happen sometimes. What I noticed is that it seems to happen more often after not using Nextcloud (the app directly or the DocumentsProvider) for a longer time or after reboots of my phone. When I try it a second time after it initially failed it almost always works without problems.

One thing I noticed is this specific line in the logcat (full error below):
```
android.os.ParcelableException: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org
```
which seem to indicate that the path for the temporary folder in
https://github.com/nextcloud/android/blob/5f38da12716d1f7dd45899507ce1e7d6ac28b8ef/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.java#L482
does not get constructed properly (notice the `/null/` in the beginning of the path.
Could it be that the Nextcloud app is not fully initialized yet when that function is called? There are a lot of initialization messages **after** that error already happened in the log.

### Steps to reproduce
1. Perform a backup in andOTP (maybe this also happens with other apps as well) to a storage location provided by the Nextcloud DocumentsProvider
2. File does not get created and the error below is shown in the logcat

### Expected behaviour
- File gets created and uploaded correctly

### Actual behaviour
- An error occurs preventing the file from being created

### Can you reproduce this problem on https://try.nextcloud.com?
- Problem is with the Android app, not the server

### Environment data
Android version: 10

Device model: OnePlus Nord

Stock or customized system: LineageOS

Nextcloud app version: 3.15.1 (F-Droid)

Nextcloud server version: 21.0.0

Reverse proxy: nginx

#### Android logcat
```
2021-03-23 07:00:37.302 1882-1913/system_process I/ActivityManager: Start proc 4837:com.nextcloud.client/u0a108 for content provider {com.nextcloud.client/com.owncloud.android.providers.DocumentsStorageProvider}
2021-03-23 07:00:37.304 4837-4837/? W/extcloud.clien: Unexpected CPU variant for X86 using defaults: x86_64
2021-03-23 07:00:37.332 4712-4743/org.shadowice.flocke.andotp.dev D/EGL_emulation: eglMakeCurrent: 0x759d61feae20: ver 2 0 (tinfo 0x759dacc247e0)
2021-03-23 07:00:37.363 2367-4012/com.android.inputmethod.latin D/EGL_emulation: eglMakeCurrent: 0x759dad269d40: ver 2 0 (tinfo 0x759dad2188a0)
2021-03-23 07:00:37.378 4837-4837/com.nextcloud.client I/extcloud.clien: The ClassLoaderContext is a special shared library.
2021-03-23 07:00:37.391 4837-4837/com.nextcloud.client I/MultiDex: VM with version 2.1.0 has multidex support
2021-03-23 07:00:37.391 4837-4837/com.nextcloud.client I/MultiDex: Installing application
2021-03-23 07:00:37.391 4837-4837/com.nextcloud.client I/MultiDex: VM has multidex support, MultiDex support library is disabled.
2021-03-23 07:00:37.473 4837-4858/com.nextcloud.client D/DocumentsStorageProvider: queryChildDocuments(), id=-1586834217/18
2021-03-23 07:00:37.508 4837-4873/com.nextcloud.client D/DocumentsStorageProvider: run ReloadFolderDocumentTask(), id=-1586834217/18
2021-03-23 07:00:37.516 4837-4873/com.nextcloud.client D/NetworkUtils: Searching known-servers store at /data/user/0/com.nextcloud.client/files/knownServers.bks
2021-03-23 07:00:37.517 4837-4873/com.nextcloud.client D/NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
2021-03-23 07:00:37.519 4837-4858/com.nextcloud.client D/DocumentsStorageProvider: createDocument(), id=-1586834217/18
2021-03-23 07:00:37.520 4837-4858/com.nextcloud.client E/DatabaseUtils: Writing exception to parcel
    android.os.ParcelableException: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org
        at android.provider.DocumentsProvider.call(DocumentsProvider.java:1069)
        at android.content.ContentProvider.call(ContentProvider.java:2152)
        at android.content.ContentProvider$Transport.call(ContentProvider.java:477)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:277)
        at android.os.Binder.execTransactInternal(Binder.java:1021)
        at android.os.Binder.execTransact(Binder.java:994)
     Caused by: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org
        at com.owncloud.android.providers.DocumentsStorageProvider.createFile(DocumentsStorageProvider.java:485)
        at com.owncloud.android.providers.DocumentsStorageProvider.createDocument(DocumentsStorageProvider.java:439)
        at android.provider.DocumentsProvider.callUnchecked(DocumentsProvider.java:1121)
        at android.provider.DocumentsProvider.call(DocumentsProvider.java:1067)
        at android.content.ContentProvider.call(ContentProvider.java:2152) 
        at android.content.ContentProvider$Transport.call(ContentProvider.java:477) 
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:277) 
        at android.os.Binder.execTransactInternal(Binder.java:1021) 
        at android.os.Binder.execTransact(Binder.java:994) 
2021-03-23 07:00:37.522 4712-4836/org.shadowice.flocke.andotp.dev W/DocumentsContract: Failed to create document
    android.os.ParcelableException: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org
        at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:82)
        at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:79)
        at android.os.Parcel.readParcelable(Parcel.java:2973)
        at android.os.Parcel.createException(Parcel.java:2066)
        at android.os.Parcel.readException(Parcel.java:2039)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:188)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
        at android.content.ContentProviderProxy.call(ContentProviderNative.java:658)
        at android.content.ContentResolver.call(ContentResolver.java:2042)
        at android.provider.DocumentsContract.createDocument(DocumentsContract.java:1327)
        at androidx.documentfile.provider.TreeDocumentFile.createFile(TreeDocumentFile.java:53)
        at androidx.documentfile.provider.TreeDocumentFile.createFile(TreeDocumentFile.java:45)
        at org.shadowice.flocke.andotp.Utilities.BackupHelper.backupFile(BackupHelper.java:49)
        at org.shadowice.flocke.andotp.Tasks.GenericBackupTask.doInBackground(GenericBackupTask.java:37)
        at org.shadowice.flocke.andotp.Tasks.GenericBackupTask.doInBackground(GenericBackupTask.java:15)
        at org.shadowice.flocke.andotp.Tasks.UiBasedBackgroundTask.runTask(UiBasedBackgroundTask.java:69)
        at org.shadowice.flocke.andotp.Tasks.UiBasedBackgroundTask.lambda$axpKRiqIDIp485gDKX0d3nZRrcg(Unknown Source:0)
        at org.shadowice.flocke.andotp.Tasks.-$$Lambda$UiBasedBackgroundTask$axpKRiqIDIp485gDKX0d3nZRrcg.run(Unknown Source:2)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.os.ParcelableException.readFromParcel(ParcelableException.java:56)
        at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:82) 
        at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:79) 
        at android.os.Parcel.readParcelable(Parcel.java:2973) 
        at android.os.Parcel.createException(Parcel.java:2066) 
        at android.os.Parcel.readException(Parcel.java:2039) 
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:188) 
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140) 
        at android.content.ContentProviderProxy.call(ContentProviderNative.java:658) 
        at android.content.ContentResolver.call(ContentResolver.java:2042) 
        at android.provider.DocumentsContract.createDocument(DocumentsContract.java:1327) 
        at androidx.documentfile.provider.TreeDocumentFile.createFile(TreeDocumentFile.java:53) 
        at androidx.documentfile.provider.TreeDocumentFile.createFile(TreeDocumentFile.java:45) 
        at org.shadowice.flocke.andotp.Utilities.BackupHelper.backupFile(BackupHelper.java:49) 
        at org.shadowice.flocke.andotp.Tasks.GenericBackupTask.doInBackground(GenericBackupTask.java:37) 
        at org.shadowice.flocke.andotp.Tasks.GenericBackupTask.doInBackground(GenericBackupTask.java:15) 
        at org.shadowice.flocke.andotp.Tasks.UiBasedBackgroundTask.runTask(UiBasedBackgroundTask.java:69) 
        at org.shadowice.flocke.andotp.Tasks.UiBasedBackgroundTask.lambda$axpKRiqIDIp485gDKX0d3nZRrcg(Unknown Source:0) 
        at org.shadowice.flocke.andotp.Tasks.-$$Lambda$UiBasedBackgroundTask$axpKRiqIDIp485gDKX0d3nZRrcg.run(Unknown Source:2) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
2021-03-23 07:00:37.532 4837-4873/com.nextcloud.client D/OwnCloudClient #0: Creating OwnCloudClient
2021-03-23 07:00:37.532 4837-4873/com.nextcloud.client D/AccountUtils: Restoring cookies for flocke@cloud.shadowice.org
2021-03-23 07:00:37.533 4837-4873/com.nextcloud.client D/AccountUtils: Restoring cookies for flocke@cloud.shadowice.org
2021-03-23 07:00:37.540 4837-4873/com.nextcloud.client D/RefreshFolderOperation: Checking changes in flocke@cloud.shadowice.org/andOTP/Test/
2021-03-23 07:00:37.544 4837-4873/com.nextcloud.client W/System.err: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
2021-03-23 07:00:37.544 4837-4873/com.nextcloud.client W/System.err: SLF4J: Defaulting to no-operation (NOP) logger implementation
2021-03-23 07:00:37.544 4837-4873/com.nextcloud.client W/System.err: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
2021-03-23 07:00:37.546 4837-4837/com.nextcloud.client V/NativeCrypto: Registering org/conscrypt/NativeCrypto's 288 native methods...
2021-03-23 07:00:37.562 4837-4837/com.nextcloud.client I/MainApp: Using Conscrypt/2.5.1 for TLS
2021-03-23 07:00:37.570 4837-4837/com.nextcloud.client I/MainApp: Enabled protocols: [TLSv1, TLSv1.1, TLSv1.2, TLSv1.3] }
2021-03-23 07:00:37.570 4837-4837/com.nextcloud.client I/MainApp: Enabled ciphers: [TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_EMPTY_RENEGOTIATION_INFO_SCSV] }
2021-03-23 07:00:37.576 4837-4837/com.nextcloud.client I/MainApp: Started 0 migrations
2021-03-23 07:00:37.577 4837-4837/com.nextcloud.client E/DisplayUtils: Failed to use reflection to enable proper vector scaling
2021-03-23 07:00:37.579 1631-1751/? D/gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 331776
2021-03-23 07:00:37.585 1631-1751/? D/gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 331776
2021-03-23 07:00:37.586 4837-4875/com.nextcloud.client D/skia: --- Failed to create image decoder with message 'unimplemented'
2021-03-23 07:00:37.587 4837-4875/com.nextcloud.client D/skia: --- Failed to create image decoder with message 'unimplemented'
2021-03-23 07:00:37.587 4712-4743/org.shadowice.flocke.andotp.dev D/EGL_emulation: eglMakeCurrent: 0x759d61feae20: ver 2 0 (tinfo 0x759dacc247e0)
2021-03-23 07:00:37.587 4837-4875/com.nextcloud.client D/ThumbnailsCacheManager: thumbnail cache dir: /data/user/0/com.nextcloud.client/cache/thumbnailCache
2021-03-23 07:00:37.590 4837-4837/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:37.591 1631-1751/? D/gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 331776
2021-03-23 07:00:37.593 4712-4743/org.shadowice.flocke.andotp.dev D/EGL_emulation: eglMakeCurrent: 0x759d61feae20: ver 2 0 (tinfo 0x759dacc247e0)
2021-03-23 07:00:37.594 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:37.602 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:37.610 4837-4877/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@6791480
2021-03-23 07:00:37.611 4837-4877/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:37.621 4837-4837/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:37.624 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:37.624 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:37.625 4837-4837/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:37.627 4837-4873/com.nextcloud.client D/OwnCloudClient #0: REQUEST PROPFIND /remote.php/webdav/andOTP/Test/
2021-03-23 07:00:37.629 4837-4878/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@1db8d0a
2021-03-23 07:00:37.629 4837-4878/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:37.629 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:37.630 4837-4837/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:37.636 4837-4877/com.nextcloud.client I/ServerNameIndicator: SSLSocket implementation: com.android.org.conscrypt.Java8FileDescriptorSocket
2021-03-23 07:00:37.637 4837-4877/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:37.639 4837-4878/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:37.643 4837-4879/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@31b144
2021-03-23 07:00:37.643 4837-4879/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:37.644 4837-4879/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:37.647 4837-4873/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@b32e54f
2021-03-23 07:00:37.647 4837-4873/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 5000 and socket timeout 40000
2021-03-23 07:00:37.690 4837-4873/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:37.712 4837-4884/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:37.717 4837-4884/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:37.717 4837-4884/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:37.724 4837-4885/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@aa02f47
2021-03-23 07:00:37.725 4837-4885/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:37.726 4837-4885/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:37.731 4837-4872/com.nextcloud.client I/WM-WorkerWrapper: Worker result SUCCESS for Work [ id=2a783710-a712-474e-ad59-8ee58cebdfb2, tags={ name:periodic_files_sync, com.nextcloud.client.jobs.FilesSyncWork, timestamp:1616479237588, * } ]
2021-03-23 07:00:37.761 4837-4886/com.nextcloud.client D/MediaProvider: Reading images for null
2021-03-23 07:00:37.800 4837-4886/com.nextcloud.client V/ArbitraryDataProvider: Updating arbitrary data with cloud id: global key: media_folders value: {"imageMediaFolders":["/storage/emulated/0"],"videoMediaFolders":[]}
2021-03-23 07:00:37.807 4837-4868/com.nextcloud.client I/WM-WorkerWrapper: Worker result SUCCESS for Work [ id=b177ce2f-8f83-4c42-b9da-3e1bfb8f8c20, tags={ com.nextcloud.client.jobs.MediaFoldersDetectionWork, *, name:immediate_media_folder_detection, timestamp:1616479237619 } ]
2021-03-23 07:00:38.022 4837-4879/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:38.022 4837-4877/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:38.024 4837-4878/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:38.035 4837-4879/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:38.035 4837-4879/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:38.037 4837-4877/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:38.038 4837-4877/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:38.045 4837-4878/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:38.045 4837-4878/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:38.049 4837-4879/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@49282b8
2021-03-23 07:00:38.050 4837-4879/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:38.050 4837-4879/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:38.051 4837-4877/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@f0663f6
2021-03-23 07:00:38.051 4837-4877/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:38.052 4837-4877/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:38.053 4837-4878/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@3491264
2021-03-23 07:00:38.053 4837-4878/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:38.053 4837-4878/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:38.065 4837-4873/com.nextcloud.client I/RefreshFolderOperation: Checked flocke@cloud.shadowice.org/andOTP/Test/ : changed
2021-03-23 07:00:38.067 4837-4873/com.nextcloud.client D/OwnCloudClient #0: REQUEST PROPFIND /remote.php/webdav/andOTP/Test/
2021-03-23 07:00:38.128 4837-4885/com.nextcloud.client D/UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-03-23 07:00:38.131 4837-4885/com.nextcloud.client V/UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-03-23 07:00:38.131 4837-4885/com.nextcloud.client V/UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
2021-03-23 07:00:38.138 4837-4885/com.nextcloud.client D/AdvancedSslSocketFactory: Creating SSL Socket with remote cloud.shadowice.org:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@2418c9
2021-03-23 07:00:38.138 4837-4885/com.nextcloud.client D/AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
2021-03-23 07:00:38.140 4837-4885/com.nextcloud.client I/ServerNameIndicator: SNI done, hostname: cloud.shadowice.org
2021-03-23 07:00:38.350 4837-4873/com.nextcloud.client I/ReadFolderRemoteOperation: Synchronized /andOTP/Test/: Operation finished with HTTP status code 207 (success)
2021-03-23 07:00:38.350 4837-4873/com.nextcloud.client D/RefreshFolderOperation: Synchronizing flocke@cloud.shadowice.org/andOTP/Test/
2021-03-23 07:00:38.355 4837-4873/com.nextcloud.client D/RefreshFolderOperation: Remote folder /andOTP/Test/ changed - starting update of local data 
2021-03-23 07:00:38.374 4837-4873/com.nextcloud.client D/FileDataStorageManager: Saving folder /andOTP/Test/ with 0 children and 0 files to remove
2021-03-23 07:00:38.375 4837-4873/com.nextcloud.client D/FileDataStorageManager: Sending 1 operations to FileContentProvider
2021-03-23 07:00:38.375 4837-4873/com.nextcloud.client D/FileContentProvider: applying batch in provider com.owncloud.android.providers.FileContentProvider@ce249fc (temporary: false)
2021-03-23 07:00:38.385 4837-4873/com.nextcloud.client D/FileContentProvider: applied batch in provider com.owncloud.android.providers.FileContentProvider@ce249fc
2021-03-23 07:00:38.405 4837-4873/com.nextcloud.client D/RefreshFolderOperation: Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_CONTENTS_SYNCED
2021-03-23 07:00:38.405 4837-4873/com.nextcloud.client D/RefreshFolderOperation: Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
2021-03-23 07:00:39.334 4837-4852/com.nextcloud.client W/System: A resource failed to call close. 
2021-03-23 07:00:39.350 4837-4852/com.nextcloud.client I/chatty: uid=10108(com.nextcloud.client) FinalizerDaemon identical 14 lines
2021-03-23 07:00:39.350 4837-4852/com.nextcloud.client W/System: A resource failed to call close. 
```

#### Web server error log
```
No errors on webserver
```

#### Nextcloud log (data/nextcloud.log)
```
No errors in nextcloud.log
```

**Comment by tobiasKaminsky** [comment]
> Caused by: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/flocke@cloud.shadowice.org

This null is strange…do you know where this comes from?

**Comment by flocke** [comment]
The path seems to get built here: https://github.com/nextcloud/android/blob/5f38da12716d1f7dd45899507ce1e7d6ac28b8ef/src/main/java/com/owncloud/android/utils/FileStorageUtils.java#L104

But for some reason `MainApp.getStoragePath()` seems to return `null`. My guess is that there is still some initialization going on and the variable that the `MainApp` is supposed to return is not yet set at the time it is called. But I have no idea why.

**Comment by tmo1** [comment]
Hello,

I'm the developer of [SMS Import / Export](https://github.com/tmo1/sms-ie), and a user of my app [has just encountered what seems to be this same issue](https://github.com/tmo1/sms-ie/issues/49). Should I open a new issue, or keep the disscussion here?

**Comment by AlvaroBrey** [comment]
> Hello,
> 
> I'm the developer of [SMS Import / Export](https://github.com/tmo1/sms-ie), and a user of my app [has just encountered what seems to be this same issue](https://github.com/tmo1/sms-ie/issues/49). Should I open a new issue, or keep the disscussion here?

Reopened, as this seems to have been wrongly closed by the stale bot. Keep the discussion here

**Comment by tmo1** [comment]
Okay. I tried reproducing the reporter's problem with a Nextcloud setup of my own, and I also got a crash, but a different one, due to a `NetworkOnMainThreadException`. As per [here](https://github.com/nextcloud/android/issues/7350) and [here](https://github.com/nextcloud/android/issues/8028), I moved the DocumentsProvider access off the Main thread, and now it works correctly. I'm not seeing the user's problem, so I'll have to check whether he's using the latest versions of Nextcloud and the Nextcloud app.
